src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1585,6 +1587,45 @@ public void unpublishFeedbackSession(String feedbackSessionName,
             throw new RuntimeException(ERROR_SENDING_EMAILS, e);
         }
     }
+    
+    public EmailWrapper sendConfirmationEmailForSubmission(String courseId, String feedbackSessionName,
+                                                           String userId, String unregisteredStudentEmail,
+                                                           String regKey)
+                    throws EntityDoesNotExistException {
+        
+        if (!isFeedbackSessionExists(feedbackSessionName, courseId)) {
+            throw new EntityDoesNotExistException(ERROR_NON_EXISTENT_FS_CONFIRM + courseId + "/" + feedbackSessionName);
+        }
+        FeedbackSessionAttributes session = getFeedbackSession(feedbackSessionName, courseId);
+        StudentAttributes student = null;
+        InstructorAttributes instructor = null;
+        
+        if (userId != null) {
+            student = studentsLogic.getStudentForCourseIdAndGoogleId(courseId, userId);
+            instructor = instructorsLogic.getInstructorForGoogleId(courseId, userId);
+        }
+        
+        if (student == null && unregisteredStudentEmail != null) {
+            student = new StudentAttributes();
+            student.email = unregisteredStudentEmail;
+            student.name = unregisteredStudentEmail;
+            student.key = regKey;
+        }
+        
+        Assumption.assertFalse(student == null && instructor == null);
+        
+        try {
+            EmailWrapper email = instructor == null
+                    ? new EmailGenerator().generateFeedbackSubmissionConfirmationEmailForStudent(session,
+                            student, TimeHelper.formatTime12H(Calendar.getInstance().getTime()))
+                    : new EmailGenerator().generateFeedbackSubmissionConfirmationEmailForInstructor(session,
+                            instructor, TimeHelper.formatTime12H(Calendar.getInstance().getTime()));
```

Comment:
`TimeHelper.formatTime12H(Calendar.getInstance().getTime())` is defined twice, extract it to a local variable.